### DISCUSSION TITLE
how can merge a json file / object / array to repeat keys with different values?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#63950] [Help] [answered]
### Summary

For example in a json file I have an array of objects, looking kind of like this:

```
[ 
{"id": 3,
   
    "Contact": " The guy",
    "Part": "abc",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",
  },

{"id": 1,
   
    "Contact": " The guy"
    "Part": "xyz",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",
  },
{"id": 2,
    
    "Contact": " The other guy",
    "Part": "xyz",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",
  },
]
```

^^^  In this array I have 3 objects,  2 of the objects are quotes for "the guy" with "Part" of abc and xyz.   The  3rd object is a  different quote for "the other guy"

I do not want to do an array of just Part, because it isn't very readable and would look like abc, xyz together..

Somehow I would like to merge the objects that are directed to a single person, but I believe I can't have repeated keys for an object?  So, what would be the best way to do it?

ideally... something similar to this below.??

```
[ 
{"id": 3,
    
    "Contact": " The guy",
    "Part": "abc",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",

    "Part": "xyz",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",
  },

///  ^^  If you notice above Part, Qty, Price, ExtPrice are all repeated keys with different value on the Part value.

{"id": 2,
   
    "Contact": " The other guy",
    "Part": "xyz",
    "Qty": 2,
    "Price": "$20.00",
    "ExtPrice": "$40.00",
  }
]
```

I would like to catch the Contact value and put it somewhere, while the other repeated keys maybe catch them in a string and list them line by line?

What would be a good way to do this?

**Comment by icyJoseph** [comment, marked as answer]
I guess, you have a few options at hand. 

- A nested property within each entry, say `quotes`:
```json
[
   {
      "id":3,
      "Contact":" The guy",
      "Quotes":[
         {
            "Part":"abc",
            "Qty":2,
            "Price":"$20.00",
            "ExtPrice":"$40.00"
         },
         {
            "Part":"xyz",
            "Qty":2,
            "Price":"$20.00",
            "ExtPrice":"$40.00"
         }
      ]
   },
   {
      "id":2,
      "Contact":" The other guy",
      "Quotes":[
         {
            "Part":"xyz",
            "Qty":2,
            "Price":"$20.00",
            "ExtPrice":"$40.00"
         }
      ]
   }
]
```

You could do this modification before saving the JSON, perhaps?

- Write a sort of parser that, without modifying your original JSON, walks the list of entries, and groups stuff by a given key. This is actually available in the JS standard nowadays, but you have to have the right browser/node.js version. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy


I guess there are other ways, but the root issue is having a proper data structure, or having the right methods to consume the chosen data structure.

**Reply by Elindo586** [reply]
Thanks... nesting the information in the object works...
